Chapter3 / HW 12

RoelofWobben opened the pull request.

RoelofWobben committed:
solved this one

Almantask commented on Src/BootCamp.Chapter/Address.cs:
```diff
@@ -8,17 +8,44 @@ public class Address
+
+            var splittedAdress = addressString.Split(Environment.NewLine);
+
+            if (splittedAdress.Length != 7)
```
7 is a magic number.

RoelofWobben replied:
oke, so I have to make a constant named totalMandotoryProperties which holds 7 ?

Almantask commented on Src/BootCamp.Chapter/Address.cs:
```diff
@@ -8,17 +8,44 @@ public class Address
+
+        public static bool operator ==(Address address1, Address address2)
+        {
+            return (address1.Recipient == address2.Recipient) && (address1.Street == address2.Street) && (address1.PostalCode == address2.PostalCode); 
```
Be more declarative:
condition
&& anotherCondition
&& oneMoreCondition
(for 3+ conditions)

RoelofWobben replied:
oke, you mean that I have to put every condition on a own line.

Almantask reviewed (approved):
Looks good, well done :)

Almantask commented:
What you have looks fine, however you are missing a part of an exercise:
```
Find which postal office (by postal code) had the biggest amount of duplicates.
```

ghost commented:
I did that in main.cs : 

```
var data = ParseFile.Parse();
            var foundZipCodes = Statics.FindPostOfficeWithMostErrors(data);
            Console.WriteLine(foundZipCodes);
```

The console.WriteLine give me the answer to the question.

RoelofWobben committed:
- added a isEqual and hash method.
- made a const for the magic number
_ make the long line in == more readable to place every condition on a own line

Almantask commented on Src/BootCamp.Chapter/Statics.cs:
```diff
@@ -0,0 +1,28 @@
+
+namespace BootCamp.Chapter
+{
+    internal class Statics
```
Do not name classes semantically, name them logically.
This one is: PostOffice

Almantask commented on Src/BootCamp.Chapter/Statics.cs:
```diff
@@ -0,0 +1,28 @@
+{
+    internal class Statics
+    {
+        internal static void FindPostOfficeWithMostErrors(List<Address> data)
```
If the above is PostOffice, then the method becomes:
PostOffice.FindMostErrors(data)
